core/src/main/java/com/linecorp/armeria/common/stream/FilteredStreamMessage.java

```diff
@@ -141,18 +143,16 @@ public final void subscribe(Subscriber<? super U> subscriber, EventExecutor exec
 
     private void subscribe(Subscriber<? super U> subscriber, EventExecutor executor, boolean withPooledObjects,
                            boolean notifyCancellation) {
-        upstream.subscribe(new FilteringSubscriber(subscriber, withPooledObjects),
-                           executor, filteringSubscriptionOptions(notifyCancellation));
+        upstream.subscribe(new FilteringSubscriber(subscriber, withPooledObjects, notifyCancellation),
+                           executor, filteringSubscriptionOptions());
     }
 
-    private SubscriptionOption[] filteringSubscriptionOptions(boolean notifyCancellation) {
+    private SubscriptionOption[] filteringSubscriptionOptions() {
         final ArrayList<SubscriptionOption> list = new ArrayList<>(2);
+        list.add(SubscriptionOption.NOTIFY_CANCELLATION);
         if (filterSupportsPooledObjects) {
             list.add(SubscriptionOption.WITH_POOLED_OBJECTS);
         }
-        if (notifyCancellation) {
-            list.add(SubscriptionOption.NOTIFY_CANCELLATION);
-        }
         return list.toArray(EMPTY_OPTIONS);
```

Comment:
Could we use static values for this? For example:
```java
private static final SubscriptionOption[] CANCELLATION_OPTION = { SubscriptionOption.NOTIFY_CANCELLATION };

if (filterSupportsPooledObjects) {
    return SubscriptionOption.values();
} else {
    return CANCELLATION_OPTION;
}
```

Reply:
Let me just use it without making the array. 😄